src/librustc_codegen_llvm/abi.rs

```diff
@@ -266,7 +266,8 @@ impl ArgTypeExt<'ll, 'tcx> for ArgType<'tcx, Ty<'tcx>> {
                 OperandValue::Ref(next(), Some(next()), self.layout.align.abi).store(bx, dst);
             }
             PassMode::Direct(_) | PassMode::Indirect(_, None) | PassMode::Cast(_) => {
-                self.store(bx, next(), dst);
```

Comment:
I think it will be better, for historical reference at least, if you factor out the changes that are to *accommodate* the restriction (vs the changes that implement the restriction) into a separate commit.